cli/lib/tasks/process-js-task.js

```diff
@@ -0,0 +1,381 @@
+const appc = require('node-appc');
+const { IncrementalFileTask } = require('appc-tasks');
+const crypto = require('crypto');
+const fs = require('fs-extra');
+const jsanalyze = require('node-titanium-sdk/lib/jsanalyze');
+const path = require('path');
+const pLimit = require('p-limit');
+const { promisify } = require('util');
+
+const i18n = appc.i18n(__dirname);
+const __ = i18n.__;
+const limit = pLimit(8);
```

Comment:
Why we need to limit the promises at all? Are we not slowing things down artificially? Or is there some reason we need to do this? If so, why 8 as a limit?